### PR TITLE
feat(gateway-api): initial support

### PR DESCRIPTION
**Pull request by bishalbera**
Closes [#764](https://github.com/glasskube/glasskube/issues/764)

### Description

Adding a new package called gateway-api. there is a problem.  maybe yaml parser is failing to to parse the ```---``` properly between comments so first that should be solved.
Screenshot 2024-07-03 at 5 49 40 PM

### CODE REVIEW

**Comment by bishalbera** [comment]
@christophenne So, now can I work on a different issue?

**Comment by christophenne** [comment]
> @christophenne So, now can I work on a different issue?

If you want, you can of course also try to debug the operator and check whether the redirect problem is the actual problem (that was just a guess by me). Maybe it's also something else.

**Comment by christophenne** [comment]
This is blocked by https://github.com/glasskube/glasskube/issues/952

File: packages/gateway-api/v1.1.0/package.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
...
+shortDescription: The Gateway API provides a standardized way to configure and manage network traffic routing in Kubernetes clusters.
+longDescription: >-
+  The Gateway API is a Kubernetes extension that standardizes the configuration and management of network traffic routing. It introduces new resource types like Gateway, GatewayClass, and HTTPRoute to provide flexible and extensible traffic control. This API enables better separation of concerns and more robust traffic management in Kubernetes environments.
+iconUrl: "https://avatars.githubusercontent.com/u/36015203"
+defaultNamespace: "gateway-api-system"
+manifests:
+  - url: https://objects.githubusercontent.com/github-production-release-asset-2e65be/219058535/32e6db11-1a9d-4159-9f74-ac18d09024ce?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240618%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240618T130009Z&X-Amz-Expires=300&X-Amz-Signature=bb468caa1b2889c387276679659b6f95c70b6d7c2350621aa885b8c402959707&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=219058535&response-content-disposition=attachment%3B%20filename%3Dstandard-install.yaml&response-content-type=application%2Foctet-stream
```
Comment (christophenne) [inline packages/gateway-api/v1.1.0/package.yaml:10]: Please change the manifest url back to `https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.1.0/standard-install.yaml`

File: packages/gateway-api/v1.1.0/package.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# yaml-language-server: $schema=https://glasskube.dev/schemas/v1/package-manifest.json
+
+name: "gateway-api"
+shortDescription: The Gateway API provides a standardized way to configure and manage network traffic routing in Kubernetes clusters.
+longDescription: >-
+  The Gateway API is a Kubernetes extension that standardizes the configuration and management of network traffic routing. It introduces new resource types like Gateway, GatewayClass, and HTTPRoute to provide flexible and extensible traffic control. This API enables better separation of concerns and more robust traffic management in Kubernetes environments.
+iconUrl: "https://avatars.githubusercontent.com/u/36015203"
+defaultNamespace: "gateway-api-system"
```
Comment (christophenne) [inline packages/gateway-api/v1.1.0/package.yaml:8]: Not sure about this, it seems to me that this package only installs CRDs right? In that case, I think the `defaultNamespace` can be removed. I also tried installing the package, the namespace is not being created.

File: packages/index.yaml [hunk 3]
```diff
@@ -48,3 +48,7 @@ packages:
     latestVersion: v2.15.3+1
     name: sealed-secrets
     shortDescription: A Kubernetes controller and tool for one-way encrypted Secrets
+  - iconUrl: "https://avatars.githubusercontent.com/u/36015203"
```
Comment (christophenne) [inline packages/index.yaml:51]: Please remove the changes to this file, as it will be updated automatically by our pipeline.

**Review by christophenne — CHANGES_REQUESTED** [review]

**Comment by bishalbera** [comment]
@christophenne made those changes . Check now please

**Comment by christophenne** [comment]
> @christophenne made those changes . Check now please

Thanks, one last thing that would be nice is if you could squash the two commits and change the message to `feat(gateway-api): initial support`, since we don't have the Github merge queue in this repository. Other than that it looks good to me.

**Comment by bishalbera** [comment]
@christophenne Please check is it ok?

**Comment by christophenne** [comment]
> @christophenne Please check is it ok?

yes, thank you

**Comment by bishalbera** [comment]
> > @christophenne Please check is it ok?
> 
> yes, thank you

ok, Nice

**Review by christophenne — APPROVED** [review]

**Comment by kosmoz** [comment]
Maybe we should extend the description to clarify that this package only includes the CRDs and a compatible controller must be installed separately? I think otherwise it might be confusing to someone now knowing what this is…

File: packages/gateway-api/v1.1.0/package.yaml
Comment (kosmoz) [inline packages/gateway-api/v1.1.0/package.yaml:1]: Please add a default namespace. It doesn't do anything for this package but it is required by the schema

**Review by kosmoz — CHANGES_REQUESTED** [review]

File: packages/gateway-api/v1.1.0/package.yaml
Comment (bishalbera) [reply packages/gateway-api/v1.1.0/package.yaml:1]: @kosmoz added check now please .

File: packages/gateway-api/v1.1.0/package.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# yaml-language-server: $schema=https://glasskube.dev/schemas/v1/package-manifest.json
+
+name: "gateway-api"
+shortDescription: The Gateway API provides a standardized way to configure and manage network traffic routing in Kubernetes clusters.
+longDescription: >-
+  The Gateway API is a Kubernetes extension that standardizes the configuration and management of network traffic routing. It introduces new resource types like Gateway, GatewayClass, and HTTPRoute to provide flexible and extensible traffic control. This API enables better separation of concerns and more robust traffic management in Kubernetes environments.
```
Comment (kosmoz) [inline packages/gateway-api/v1.1.0/package.yaml:6]:
```suggestion
  The Gateway API is a Kubernetes extension that standardizes the configuration and management of network traffic routing.
  It introduces new resource types like Gateway, GatewayClass, and HTTPRoute to provide flexible and extensible traffic control.
  This API enables better separation of concerns and more robust traffic management in Kubernetes environments.

  > **Note:** This package installs the Gateway API CRDs from the official source.
    To utilize them, install a controller implementing this API.
    You can find a list of implementations [here](https://gateway-api.sigs.k8s.io/implementations/).
```

**Review by kosmoz — CHANGES_REQUESTED** [review]

**Comment by kosmoz** [comment]
Note: v0.13.0 has been released, which includes the fix required for this package.

https://github.com/glasskube/glasskube/releases/tag/v0.13.0

**Comment by bishalbera** [comment]
> Note: v0.13.0 has been released, which includes the fix required for this package.
> 
> https://github.com/glasskube/glasskube/releases/tag/v0.13.0

Ya I saw that . Thanks :)

**Review by kosmoz — APPROVED** [review]

**Pull request merged**